### PR TITLE
SDIT-1623 Add CSIP Factors and additional CSIP Report mappings

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/CSIPReport.kt [hunk 1]
```diff
@@ -92,6 +94,54 @@ data class CSIPReport(
...
+  @Convert(converter = YesNoConverter::class)
+  val proActiveReferral: Boolean = false,
+
+  @Column(name = "RFR_STAFF_ASSAULTED")
+  @Convert(converter = YesNoConverter::class) val staffAssaulted: Boolean = false,
+
+  @Column(name = "RFR_STAFF_NAME")
+  val staffAssaultedName: String? = null,
+
+  // ------------------ Additional Referral Details ---------------------//
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/CSIPReport.kt:107]: added separators for each section - as it's a BIG table

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/csip/CSIPResource.kt [hunk 2]
```diff
@@ -194,6 +204,30 @@ data class CSIPResponse(
...
+@JsonInclude(JsonInclude.Include.NON_NULL)
+data class ReportDetails(
+  @Schema(description = "How the offender was involved")
+  val involvement: CodeDescription?,
+  @Schema(description = "Concern description")
+  val concern: String?,
+  @Schema(description = "What was the concern")
+  val knownReasons: String?,
+  @Schema(description = "What was the concern")
+  val otherInformation: String?,
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/csip/CSIPResource.kt:216]: minor copy & paste?

**Review by andymarke — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/csip/CSIPResource.kt (hunk 2 above)
Comment (karenmillermoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/csip/CSIPResource.kt:216]: oopsy = will fix that

**Pull request merged**